ios/gamma/split-view/RNSSplitViewHostController.swift

```diff
@@ -300,10 +296,6 @@ public class RNSSplitViewHostController: UISplitViewController, ReactMountingTra
         && columns.count <= maxNumberOfColumns,
       "[RNScreens] SplitView can only have from \(minNumberOfColumns) to \(maxNumberOfColumns) columns"
     )
-
-    assert(
-      columns.count == fixedColumnsCount,
-      "[RNScreens] SplitView number of columns shouldn't change dynamically")
```

Comment:
These should stay right? As violating them still leads to "bad behaviour".

Reply:
Assertions are not there to assert that things will work in **current setup**. The assertions are there to *assert* inherent invariants of the component and give us warning when we mess something up in the future. So I say, we leave them.

Reply:
it shouldn't, now we can handle nicely that behavior by recreating the entire SplitView

Reply:
https://github.com/software-mansion/react-native-screens/pull/3339/commits/6623fefec8c39bf1cb3f24416e396198d14ec51f